Basic wireframe

Beardless-sheik opened the pull request:
In this pull request: 
 - The code was scaffolded using webpack to compile assets and babel to transpile
 - The HTML template was added according to the given wireframe
 - Styles also added according to given wireframe
 - Bootstrap was installed via CDN also for responsive styling

Beardless-sheik committed:
Update README.md

- Add current descriptions and instructions to set up the project

Beardless-sheik committed:
Add github action to run linters on every pull request

Beardless-sheik committed:
Add eslint config

Beardless-sheik committed:
Add stylelint config

Beardless-sheik committed:
Add webhint config

Beardless-sheik committed:
Add babel config

Beardless-sheik committed:
Add webpack config

Beardless-sheik committed:
Create scafold 'source' files

Beardless-sheik committed:
Create npm project by initializing package.json

Beardless-sheik committed:
Add scripts to package.json

Beardless-sheik committed:
Add initial HTML

Beardless-sheik committed:
Add webhint npm package

Beardless-sheik committed:
Add strylint npm packages

Beardless-sheik committed:
Import styles into base index.js script

Beardless-sheik committed:
Update packages

Beardless-sheik committed:
Add github workflow to deploy when a push is made on main

Beardless-sheik committed:
Create package-lock.json

Beardless-sheik committed:
Add bootstrap to be included via CDN

Beardless-sheik committed:
Add webpack dev server as a dev package for hot reloading during development

Beardless-sheik committed:
Update gitignore to ignore build files in dist

Beardless-sheik committed:
Add jest for future tests and css-loader and style-loader for webpack

Beardless-sheik committed:
Add full HTML template according to wire frame

Beardless-sheik committed:
Import styles for asset packing by webpack

Beardless-sheik committed:
Add styles to match given wireframe

Beardless-sheik committed:
Update packages to deal with linting

Beardless-sheik committed:
Update github actions

Beardless-sheik committed:
fix linter errors

rustam-tolipov commented on src/index.js:
```diff
@@ -0,0 +1 @@
+import './style.css';
```
## NICE APP 👏
- [ ] You did great job with installing webpack successfully. Everything is working as expected ⚡👍. Please add some javascript functionality and make it work a little. You must use ES6 javascript like `arrow functions`, `let `and `const `varaibles. App should not work 100% yet because you will add things with APIs in the following lessons 🍀😉

Beardless-sheik replied:
Ok..........Looking into it

rustam-tolipov reviewed (changes requested):
Hi @Beardless-sheik , 

Good job so far!

### HIGHLIGHTS!

- No linters errors.💯
- Descriptive Readme.md. 💯


There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Beardless-sheik committed:
Add packages for linters

Beardless-sheik committed:
Add form client side HTML validtion

Beardless-sheik committed:
Remove static data to be added via javascript

Beardless-sheik committed:
Update index.js

- Pick HTML items from the DOM
- Add scaffolding functions
- Import functions from other modules

Beardless-sheik committed:
Create Local Storage module to manage local storage

Beardless-sheik committed:
Create module to render elements

Beardless-sheik committed:
Add function to add scores to score array

Beardless-sheik committed:
Import addScore into index.js starter file

Beardless-sheik committed:
Finish submit form function to run as an event listener function

Beardless-sheik committed:
Add event listener to load scores from local storage

Beardless-sheik committed:
Fix lintereslint errors

o-rashad reviewed (approved):
Asslam Alaikum @Beardless-sheik ,😊

Hi👋🏽, 
Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 

You nailed it !!

- Modular code.
- Easy to read and change.
- Good Documentation.

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
